Convert aws.cloudformation.Create to use Boto3

diranged opened the pull request:
This is step 1 ... use Boto3 rather than Boto. Boto3 will be required
for us to interact with CF Change Sets.

mikhail commented on kingpin/actors/aws/cloudformation.py:
```diff
@@ -91,6 +112,39 @@ class CloudFormationBaseActor(base.AWSBaseActor):
+                'ParameterValue': value,
+                'UsePreviousValue': use_previous_value
+            })
+        return boto_params
```
Could use a list comprehension:

```python
return [{'Key': k, 'Val': v, 'Prev': prev} for k,v in a.items()]
```

mikhail commented:
Wow... painful. But glad you made the effort here!

diranged commented:
@mikhail, addressed your comment and finished this up with the Delete actor. This code has been tested locally too (with integration tests), so I know it actually works! :)

diranged commented:
CloudFormation integration tests passed in our integration environment:
```
integration_01_create_stack (kingpin.actors.aws.test.integration_cloudformation.IntegrationCreate) ... ok
integration_02_create_duplicate_stack_should_fail (kingpin.actors.aws.test.integration_cloudformation.IntegrationCreate) ... ok
integration_03_delete_stack (kingpin.actors.aws.test.integration_cloudformation.IntegrationCreate) ... ok
integration_04_delete_missing_stack_should_fail (kingpin.actors.aws.test.integration_cloudformation.IntegrationCreate) ... ok
```

mikhail commented on kingpin/actors/aws/cloudformation.py:
```diff
@@ -713,6 +730,222 @@ def _update_stack(self, stack):
+        # Get the current template for the stack, and get our local template
+        # body. Make sure they're in the same form (dict).
+        existing = yield self._get_stack_template(stack['StackId'])
+        new = json.loads(self._template_body)
```
wanna use our utils here?

diranged commented:
@mikhail I think this is ready for a review -- more work will go into this actor, but this should get us moving along. The integration tests work and execute an update:

![image](https://cloud.githubusercontent.com/assets/768067/15715639/12b23414-27d4-11e6-82cc-33796c759e14.png)

diranged commented:
@mikhail , hey man somehow this got away from us... can you review?

nextdoor-ops-ro commented:
... fairly certain I've already reviewed this. Strange. Let's merge it if
it's not conflicting with anything

On Fri, Jun 24, 2016, 8:59 PM Matt W <notifications@github.com> wrote:

> @mikhail <https://github.com/mikhail> , hey man somehow this got away
> from us... can you review?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nextdoor/kingpin/pull/374#issuecomment-228508863>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/ABJIHt1KQVQwCPlHQP7QTh8KB6scvZnSks5qPKeigaJpZM4InCr5>
> .
>
